r/tests/testthat/test-csv.R

```diff
@@ -610,3 +610,15 @@ test_that("read_csv_arrow() can read sub-second timestamps with col_types T sett
   expected <- as.POSIXct(tbl$time, tz = "UTC")
   expect_equal(df$time, expected, ignore_attr = "tzone")
 })
+
+test_that("read_csv_arrow() can't read timestamps with time zone, with the col_types T setting (ARROW-15602)", {
```

Comment:
Instead of adding a test that something doesn't work, can you add a test that shows how it does work if you pass a schema?

There is a test above called "read_csv_arrow() can read timestamps", you could add an assertion to it, or add a new test near it. You can use the example from the docs addition above.

Reply:
Thank you for your review.
I think this test case is necessary to make sure that the error messages are clear to the user.
In fact, until recently the error messages were broken in the development version (ARROW-17429).

I will add a test case for passing the schema, as I think it is indeed necessary too.

Reply:
Maybe add a comment here explaining that we're just testing that the error message is meaningful